models/content.js

```diff
@@ -540,6 +540,29 @@ async function creditOrDebitTabCoins(oldContent, newContent, options = {}) {
       amountToDebit = oldContent.tabcoins - contentDefaultEarnings + userDefaultEarnings;
     } else {
       amountToDebit = userDefaultEarnings;
+      const query = {
+        text: `
+        select distinct events.originator_user_id as user_id, balance_operations.recipient_id as content_id from balance_operations
+        left join contents on contents.id = balance_operations.recipient_id
+        left join events on events.id = balance_operations.originator_id
+        where balance_operations.amount < 0 
+        and balance_operations.balance_type='content:tabcoin'
+        and balance_operations.recipient_id = $1
+        ;`,
+        values: [newContent.id],
+      };
+
+      const results = await database.query(query);
+      results.rows.forEach(async (contentBeingDeleted) => {
+        if (!contentBeingDeleted.user_id) return;
+        await balance.create({
+          balanceType: 'user:tabcoin',
+          recipientId: contentBeingDeleted.user_id,
+          amount: 1,
+          originatorType: 'event',
+          originatorId: options.eventId,
+        });
```

Comment:
```suggestion
        await balance.create(
         {
           balanceType: 'user:tabcoin',
           recipientId: contentBeingDeleted.user_id,
           amount: 1,
           originatorType: 'event',
           originatorId: options.eventId,
         },
         {
           transaction: options.transaction,
         }
        );
```

Tudo precisa ser realizado na mesma transação. 🤝

E aqui é o ponto de decisão se realmente é otimização prematura já fazer a inserção naquela mesma consulta ou se é melhor criar esses balanços separadamente para cada user_id.

Depois que desenvolver os testes mais elaborados, dá pra comparar melhor as performances das alternativas. 👍

O nome `contentBeingDeleted` não está fazendo sentido aqui. 😅

É bom colocar tudo isso dentro de uma função que é chamada ali no `else`. 😉

Reply:
@aprendendofelipe criei uma nova função no orchestrator chamada `createTabcoinForContent` que recebe Id do usúario, id do conteúdo, ip e transação, sem o limitador de tempo, e isso funcionou para os testes de upvote e downvote separados, mas, por algum motivo, quando mesclo as duas operações juntas em um único teste, o mesmo não funciona com query sugerida, sendo assim, não é adicionado o tabcoin, e mesmo testando com a antiga query ou atualizando-a, a mesma não funciona, e estou travado neste pequeno problema.